### PR TITLE
Md 03/pokemon list

### PR DESCRIPTION
**Pull request by Danny-Mus**
check

### CODE REVIEW

File: front-dex/src/app/components/caught/caught.component.html [hunk 1]
```diff
@@ -1,5 +1,4 @@
-<div class="checkbox">
-    <input id="caught" type="checkbox">
-    <label for="caught"></label>
-    <br>
-</div>
+<label class="check">
+    <input class="check__input " type="checkbox"/>
```
Comment (Danny-Mus) [inline front-dex/src/app/components/caught/caught.component.html]: space inside class tag

File: front-dex/src/app/components/caught/caught.component.html [hunk 2]
```diff
@@ -1,5 +1,4 @@
...
+    <span class="check__checkmark"></span>
+</label>
```
Comment (Danny-Mus) [inline front-dex/src/app/components/caught/caught.component.html]: newline at the end of the file?

File: front-dex/src/app/components/caught/caught.component.scss [hunk 3]
```diff
@@ -1,28 +1,31 @@
 @import '../../../assets/scss/main.scss';
```
Comment (Danny-Mus) [inline front-dex/src/app/components/caught/caught.component.scss]: Indenting of the file is not correct

File: front-dex/src/app/components/favourite/favourite.component.html [hunk 4]
```diff
@@ -1,5 +1,4 @@
-<div class="checkbox">
-    <input id="favourite" type="checkbox">
-    <label for="favourite"></label>
-    <br>
-</div>
+<label class="check">
+    <input class="check__input " type="checkbox"/>
+    <span class="check__checkmark"></span>
+</label>
```
Comment (Danny-Mus) [inline front-dex/src/app/components/favourite/favourite.component.html]: newline?

File: front-dex/src/app/components/favourite/favourite.component.scss [hunk 5]
```diff
@@ -1,38 +1,30 @@
 @import '../../../assets/scss/main.scss';
```
Comment (Danny-Mus) [inline front-dex/src/app/components/favourite/favourite.component.scss]: check indenting of the whole file

File: front-dex/src/app/components/card/card.component.scss [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+        position: absolute;
+        right: 5%;
+        z-index: 1;
+    }
+
+    &__type {
+        width: 66px;
+        display: flex;
+        justify-content: center;
+        background: #dbdbdb;
```
Comment (DaanEKR) [inline front-dex/src/app/components/card/card.component.scss]: Use color variables

File: front-dex/src/styles.scss [hunk 7]
```diff
@@ -1,2 +1,5 @@
 @import './assets/scss/main';
 
+body {
+    background: $primary-theme-background;
+}
```
Comment (Danny-Mus) [inline front-dex/src/styles.scss]: newline at the end of the file?

File: front-dex/src/app/app.component.html [hunk 8]
```diff
@@ -1,15 +1,21 @@
 <nav class="navigation">
     <ul class="navigation__list">
+        <div class="navigation__container">
```
Comment (kmensink95) [inline front-dex/src/app/app.component.html]: U dont need a extra div if you style the navigation___list correct

File: front-dex/src/app/components/card/card.component.scss [hunk 9]
```diff
@@ -0,0 +1,231 @@
+@import '../../../assets/scss/main.scss';
+
+.pokecard {
+    width: 250px;
+    height: 250px;
+    display: flex;
+    margin: 17px;
+    align-items: flex-end;
+    flex-direction: column;
+    box-shadow: 3px 3px 5px 0px rgba(0,0,0,0.75);
```
Comment (DaanEKR) [inline front-dex/src/app/components/card/card.component.scss]: Use color variabele.

File: front-dex/src/app/pages/home/home.component.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit } from '@angular/core';
+import { PokemonService } from '../../services/pokemon/get-pokemon.service';
+import { TypeService } from '../../services/pokemon/get-pokemon-type.service';
+import Pokemon from 'src/app/models/interfaces/Pokemon';
+import { Type } from 'src/app/models/interfaces/Type';
+import { Subscriber } from 'rxjs';
```
Comment (DaanEKR) [inline front-dex/src/app/pages/home/home.component.ts]: Add space

File: front-dex/src/app/pipes/pipes.module.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+import { PokemonFilterPipe } from './pokemon-filter.pipe';
+
+@NgModule({
+  imports:      [ CommonModule ],
+  declarations: [ PokemonFilterPipe  ],
```
Comment (DaanEKR) [inline front-dex/src/app/pipes/pipes.module.ts]: Delete 1 space at the end

File: front-dex/src/app/pipes/pipes.module.ts [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+  exports:      [ PokemonFilterPipe  ]
```
Comment (DaanEKR) [inline front-dex/src/app/pipes/pipes.module.ts]: Delete 1 space at the end

File: front-dex/src/app/components/card/card.component.html [hunk 13]
```diff
@@ -0,0 +1,19 @@
+<article class="pokecard" [ngClass]="setBackgroundColor(pokemon.types, 'pokecard')">
+    <section class="pokecard__header">
+        <div class="pokecard__options">
+            <app-favourite></app-favourite>
+            <app-caught class="pokecard__option-spacing"></app-caught>
+        </div>
+
+        <div class="pokecard__types">
+            <span class="pokecard__type" *ngIf="pokemon?.types[1]" [ngClass]="setBackgroundColor(pokemon?.types[1].type.name, 'pokecard__type', false)">{{ pokemon?.types[1].type.name }}</span>
```
Comment (kmensink95) [inline front-dex/src/app/components/card/card.component.html]: Do you need a span for this?

File: front-dex/src/app/components/card/card.component.html [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+            <span class="pokecard__type" *ngIf="pokemon?.types[0]" [ngClass]="setBackgroundColor(pokemon?.types[0].type.name, 'pokecard__type', false)"> {{ pokemon?.types[0].type.name }}</span>
```
Comment (kmensink95) [inline front-dex/src/app/components/card/card.component.html]: Do you need a span for this?

File: front-dex/src/app/components/card/card.component.html [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+        </div>
+    </section>
+
+    <img class="pokecard__image" (click)="getSelectedPokemon(pokemon)" src="https://pokeres.bastionbot.org/images/pokemon/{{pokemon?.id}}.png">
+    <section class="pokecard__info">
+        <span class="pokecard__item">#{{ pokemon?.id }}</span>
```
Comment (kmensink95) [inline front-dex/src/app/components/card/card.component.html]: Do you need a span for this?

File: front-dex/src/app/components/card/card.component.html [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+        <span class="pokecard__item">{{ pokemon?.name | titlecase }}</span>
```
Comment (kmensink95) [inline front-dex/src/app/components/card/card.component.html]: Do you need a span for this?

File: front-dex/src/app/components/card/card.component.scss (hunk 6 above)
Comment (kmensink95) [inline front-dex/src/app/components/card/card.component.scss]: Use a variable

File: front-dex/src/app/pages/home/home.component.html [hunk 17]
```diff
@@ -0,0 +1,16 @@
+<div class="home-container">
```
Comment (kmensink95) [inline front-dex/src/app/pages/home/home.component.html]: Why is there a div with no styling?

File: front-dex/src/app/pages/home/home.component.ts [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss']
+})
+export class HomeComponent implements OnInit {
+  pokemons;
```
Comment (kmensink95) [inline front-dex/src/app/pages/home/home.component.ts]: Type?

File: front-dex/src/app/pages/home/home.component.ts [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+  chosenPokemon: Pokemon;
+  types;
```
Comment (kmensink95) [inline front-dex/src/app/pages/home/home.component.ts]: Type?

File: front-dex/src/app/pages/home/home.component.ts [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+  result;
```
Comment (kmensink95) [inline front-dex/src/app/pages/home/home.component.ts]: Type?

File: front-dex/src/app/components/card/card.component.ts [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+})
+export class CardComponent implements OnInit {
+  @Input() pokemon: Pokemon;
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
+  setBackgroundColor(types, name, type=true) {
```
Comment (kmensink95) [inline front-dex/src/app/components/card/card.component.ts]: What kind of function :void or :observable

File: front-dex/src/app/pages/home/home.component.ts [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+  typeSelected: number;
+  constructor(private PokemonService: PokemonService, private TypeService: TypeService) {
+  }
+
+  ngOnInit() {
```
Comment (kmensink95) [inline front-dex/src/app/pages/home/home.component.ts]: Create a new function and put the function in the ngOnInit

File: front-dex/src/app/components/card/card.component.ts [hunk 23]
```diff
@@ -1,5 +1,6 @@
 import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
 import Pokemon from 'src/app/models/interfaces/Pokemon';
+import { Observable } from 'rxjs';
```
Comment (Danny-Mus) [inline front-dex/src/app/components/card/card.component.ts]: remove this if this isn't used

**Pull request merged**